server/controllers/patients/patchPatientData.js

```diff
@@ -0,0 +1,56 @@
+const { patchPatientDataByIdQuery } = require('../../database/queries');
+const { boomify, patientDataValidation } = require('../../utils');
+
+const patchPatientData = async (req, res, next) => {
+  try {
+    const {
+      firstName,
+      lastName,
+      phone,
+      email,
+      birthday,
+      diseases,
+      patientId,
+    } = await patientDataValidation.validate(
+      { ...req.body, ...req.params },
+      {
+        abortEarly: false,
+      },
+    );
+
+    const {
+      rows: [data],
+      rowCount,
+    } = await patchPatientDataByIdQuery({
+      firstName,
+      lastName,
+      phone,
+      email,
+      birthday,
+      diseases,
+      patientId,
+    });
+
+    if (rowCount === 1) {
+      return res.json({
+        statusCode: 200,
+        message: 'Updated successfully',
+        data,
+      });
+    }
+    return next(boomify(404, `There's no patient with this Id`));
```

Comment:
we are working on " error first "so try to handle the error in the if

Reply:
Since you are using handling errors with `if-return` this is the right way, even if you tried to do if-else here, `eslint` will fix it to become like this

Reply:
this will not work because the eslint rules

Reply:
I think what @AlaaShurrab meant was 
```js
    if (rowCount === 0) {
      return next(boomify(404, `There's no patient with this Id`));
    }
    return res.json({
        statusCode: 200,
        message: 'Updated successfully',
        data,
      });
```

Reply:
Yeah, but this is not adding anything new for best practice or for the performance :sweat_smile:

Reply:
Don't do it, there's no need for it

Reply:
I've already changed it ...